provider/maas/instance.go

```diff
@@ -262,3 +263,11 @@ func (mi *maas1Instance) hostname() (string, error) {
 	// A MAAS instance has its DNS name immediately.
 	return mi.maasObject.GetField("hostname")
 }
+
+func (mi *maas1Instance) displayName() (string, error) {
+	displayName, err := mi.hostname()
```

Comment:
We need to fall back to fqdn here as well.
Add a new method fqdn() to maas1Instance to get the "fqdn" attribute